frontend/src/components/Page.js

```diff
@@ -62,40 +20,39 @@ const Container = ({ contentFullSize, children }) => (
   </main>
 )
 
-const Page = ({
-  title,
-  initialQuery,
-  requireSignIn,
-  requireSignOut,
-  contentFullSize,
-  children,
-}) => (
-  <SearchProvider initialQuery={initialQuery}>
-    <Head title={title} />
-    <NavBar />
-    <Content
-      contentFullSize={contentFullSize}
-      requireSignIn={requireSignIn}
-      requireSignOut={requireSignOut}
-    >
-      {children}
-    </Content>
-  </SearchProvider>
-)
+const Page = ({ title, initialQuery, contentFullSize, children }) => {
+  const { asPath, pathname, replace } = useRouter()
+
+  useEffect(() => {
+    const browserPath = asPath.split('?')[0]
+    const doesTheBrowserURLMismatchPathname = browserPath !== pathname
+    if (
+      doesTheBrowserURLMismatchPathname &&
+      pathname !== '/login' &&
+      browserPath !== '/'
+    ) {
+      replace(pathname, null, { shallow: true })
+    }
+  }, [asPath, pathname, replace])
```

Comment:
Why is this needed? Why can they still be mismatched?

Reply:
Added docs in 08e190b.